src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use PHPUnit\Framework\Assert;
+use Symfony\Bundle\FrameworkBundle\Client;
+
+/**
+ * Ideas borrowed from Laravel Dusk's assertions.
+ *
+ * @see https://laravel.com/docs/5.7/dusk#available-assertions
+ */
+trait WebTestAssertions
+{
+    /** @var Client|null */
+    protected static $client;
+
+    public static function assertResponseIsSuccessful(): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertTrue(
+            $response->isSuccessful(),
+            "Response was expected to be successful, but actual HTTP code is {$response->getStatusCode()}."
+        );
+    }
+
+    public static function assertHttpCodeEquals(int $expectedCode): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertSame(
+            $expectedCode,
+            $code = $response->getStatusCode(),
+            "Response code [$expectedCode] does not match actual HTTP code [$code]."
+        );
+    }
+
+    public static function assertResponseRedirects(int $expectedCode = null, string $expectedLocation = null): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertTrue(
+            $response->isRedirect(),
+            "Actual response code [{$response->getStatusCode()}] does not match any valid HTTP redirection code."
+        );
+
+        if ($expectedCode) {
+            static::assertHttpCodeEquals($expectedCode);
+        }
+
+        if ($expectedLocation) {
+            Assert::assertSame(
+                $expectedLocation,
+                $location = $response->headers->get('Location'),
+                "Location [$expectedLocation] does not match actual redirection URL [$location]."
+            );
+        }
+    }
+
+    public static function assertPageTitleEquals(string $expectedTitle): void
+    {
+        self::checkCrawlerAvailable();
+
+        $titleNode = static::$client->getCrawler()->filter('title');
+
+        Assert::assertSame(1, $titleNode->count(), 'There is no <title> tag in the current page to look for.');
```

Comment:
The message may be misleading if there are two `<title>` tags. Maybe have to assertions? First check that there is at least one (with greaterThan and message `'There is no <title> tag in the current page to look for.'` and second `Assert::assertSame(1, $titleNode->count(), 'There must be exactly one <title> tag in the page.');`

Reply:
or maybe an `assertCount` will work better?